### PR TITLE
Contact group

### PR DESCRIPTION
**Pull request by khelwood**
Added ContactGroup model. LDAPGroupReader looks up contact details
for all members of a specified group in LDAP. The found people are all created
as Contacts and offered in the internal contacts selector.

### CODE REVIEW

File: spec/lib/ldap_group_reader_spec.rb [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+      context 'when the group has members' do
+        let(:member_uids) { ['alpha', 'beta', 'gamma'] }
+        let(:ldap_group_members) { member_uids.map { |uid| "uid=#{uid},#{person_base}" } }
+
+        let(:ldap_people) do
+          member_uids.map { |uid| double('ldap-person', cn: ["#{uid} mc#{uid}"], mail: ["#{uid}@place.com"]) }
+        end
+
+        let(:person_filter) {
+          Net::LDAP::Filter.eq('uid', 'alpha') | Net::LDAP::Filter.eq('uid', 'beta') | Net::LDAP::Filter.eq('uid', 'gamma')
```
Comment (emrojo) [inline spec/lib/ldap_group_reader_spec.rb]: Normally, the most common `or` operators that are in use in ruby are `or` or ` ||`.
The operator `|` is bitwise operator, it is equivalent to `|| `when operating with booleans, but differs when operating with other datatypes. For example, is not the same `3|4` than `3||4`. In this case `Net::LDAP::Filter.eq('uid', 'alpha')` is boolean, so it's not a mistake, but could give not expected results in other case.

File: spec/lib/ldap_group_reader_spec.rb [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+        lg = double('ldap-group')
+        allow(lg).to receive(member_attr).and_return(ldap_group_members)
+        lg
+      end
+
+      before do
+        expect(ldap).to receive(:search).
+          with(filter: group_filter, base: group_base, attributes: [member_attr]).
+          and_return([ldap_group])
+      end
```
Comment (khelwood) [inline spec/lib/ldap_group_reader_spec.rb]: I see what you mean. I will change the tests.

File: spec/lib/ldap_group_reader_spec.rb (hunk 1 above)
Comment (khelwood) [inline spec/lib/ldap_group_reader_spec.rb]: It is not a boolean. It is a filter. `|` combines it with other filters.

Comment (khelwood) [inline spec/lib/ldap_group_reader_spec.rb]: See http://www.rubydoc.info/gems/ruby-net-ldap/Net/LDAP/Filter#%7C-instance_method

**Pull request merged**